Updated Edit Profile page to use newer `<CWForm/>` component with newer validations + cleaned up related stale components

mzparacha opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/7935

## Description of Changes
- Updated Edit Profile page to use newer `<CWForm/>` component with newer validations
- Cleaned up related stale components
- Added CWForm/form-validation compatibility to `ReactQuillEditor`
- DRY'ed related code from other areas of the app.

## "How We Fixed It"
N/A

## Test Plan

1. Visit the edit profile page
2. CA - Verify you can perform all the edit actions as before and that there are no unexpected issues
3. Imp: during step 2, rigorously test the quill editor and its validation states to make sure there are no inconsistencies or issues as this is one of the main components.

## Deployment Plan
N/A

## Other Considerations
There is currently a bug in production that prevents users from updating their profile avatar, this will be addressed in https://github.com/hicommonwealth/commonwealth/issues/8151

mzparacha commented on packages/commonwealth/client/scripts/views/components/EditProfile/EditProfile.tsx:
```diff
@@ -0,0 +1,433 @@
+                fullWidth
+                placeholder="Enter your user name"
+                label="Username"
+                // TODO: unique username like in PersonalInformationStep?
+                // TODO: username generator like in PersonalInformationStep?
```
Will have to confirm if we want to add this logic here. This is just for enhancement and not a blocker.

In case we want to add this logic, it will be in a separate PR.

mzparacha commented on packages/commonwealth/client/scripts/views/components/EditProfile/EditProfile.tsx:
```diff
@@ -0,0 +1,433 @@
+              description="Express yourself through imagery."
+            >
+              <CWText fontWeight="medium">Add a background image </CWText>
+              {/* TODO: add option to remove existing image */}
```
Another potential enhancement, this behavior is not yet implemented.

I feel we would want this, will be creating a new ticket for this.

masvelio reviewed (commented):
Getting bug on profile page load
<img width="1673" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/14819225/7c2a7aab-9ca7-4b40-a046-4dc57bfdcc72">

<img width="608" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/14819225/047959e3-5dd1-4ae5-98db-1bb817b2041f">


Cant see anymore option to select tile vs full image

<img width="967" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/14819225/d12dae0b-1a59-4276-a25a-5e484f624b2b">

I can set my name as Anonymous, is this correct?

<img width="558" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/14819225/e4aad210-cf6a-4465-9b73-5ac5f5672ebc">


Besides, AMAZING WORK, thanks for all refactors and proactive attitude!

mzparacha commented on packages/commonwealth/client/scripts/views/components/EditProfile/EditProfile.tsx:
```diff
@@ -0,0 +1,433 @@
+  const navigate = useNavigate();
+  const { isLoggedIn } = useUserLoggedIn();
+  const [profile, setProfile] = useState<NewProfile>();
+  const [avatarUrl, setAvatarUrl] = useState();
```
this was actually the same in pre-cleanup code. Also the avatar setting logic is broken, which I plan to fix in https://github.com/hicommonwealth/commonwealth/issues/8151

mzparacha commented:
> Getting bug on profile page load
 
good catch, fixed.

> Cant see anymore option to select tile vs full image

fixed in https://github.com/hicommonwealth/commonwealth/pull/8150/commits/a0f88652a612e7b3a7f2d7f66fb7576779dc9ca8

> I can set my name as Anonymous, is this correct?

For now, yes. I only migrated code to newer standards. Would love to add the username uniqueness / random generation, more context [here](https://github.com/hicommonwealth/commonwealth/pull/8150#discussion_r1639865537)

---

Thanks for the feedback. Fixed all the issues.
